### PR TITLE
[phone number sdk] Adds environment variable AZURE_USERAGENT_OVERRIDE, that overrides x-ms-useragent on the tests

### PR DESCRIPTION
**Pull request by automagicaly**
Change needed to fix test pipelines

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClientOptionsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Communication.Pipeline;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.PhoneNumbers
+{
+    /// <summary>
+    /// Summary description for Class1
```
Comment (danielortega-msft) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClientOptionsExtensions.cs:11]: Please update this to describe the purpose of this class.

**Review by danielortega-msft — CHANGES_REQUESTED** [review]

File: sdk/communication/Azure.Communication.PhoneNumbers/src/MSUserAgentPolicy.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.PhoneNumbers
+{
+    /// <summary>
+    /// Summary description for Class1
```
Comment (whisper6284) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/MSUserAgentPolicy.cs:12]: Update summary.

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClientOptionsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    /// <summary>
+    /// This class extends the PhoneNumbersClientsOptions to be able to inject the x-ms-useragent header into the HttpPipeline
+    /// </summary>
+    internal static class PhoneNumbersClientOptionsExtensions
+    {
+        public static HttpPipeline BuildPhoneNumbersHttpPipeline(this ClientOptions options, ConnectionString connectionString)
+        {
+            var authPolicy = new HMACAuthenticationPolicy(new AzureKeyCredential(connectionString.GetRequired("accesskey")));
+            var msUserAgentPolicy = new MSUserAgentPolicy();
+            return HttpPipelineBuilder.Build(options, authPolicy, msUserAgentPolicy);
```
Comment (whisper6284) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClientOptionsExtensions.cs:19]: Double-checking that this is how it gets called in ClientOptions?

**Comment by lucasrsant** [comment]
Could you please provide a more detailed description to this pull request? What was happening, the proposed solution, etc.

**Review by lucasrsant — CHANGES_REQUESTED** [review]
Could you please provide a more detailed description to this pull request? What was happening, the proposed solution, etc.

File: sdk/communication/Azure.Communication.PhoneNumbers/src/MSUserAgentPolicy.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    /// </summary>
+    internal class MSUserAgentPolicy : HttpPipelinePolicy
+    {
+        /// <summary>
+        /// Processed message adding the HTTP header x-ms-useragent if applicable.
```
Comment (lucasrsant) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/MSUserAgentPolicy.cs:17]: nit: Processes

File: sdk/communication/Azure.Communication.PhoneNumbers/src/MSUserAgentPolicy.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        /// <param name="message"></param>
+        /// <param name="pipeline"></param>
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            addHeader(message);
+            ProcessNext(message, pipeline);
+        }
+
+        /// <summary>
+        /// Processed message asynchronously adding the HTTP header x-ms-useragent if applicable.
```
Comment (lucasrsant) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/MSUserAgentPolicy.cs:28]: nit: Processes

**Review by lucasrsant — CHANGES_REQUESTED** [review]

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClientOptionsExtensions.cs (hunk 3 above)
Comment (whisper6284) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClientOptionsExtensions.cs:19]: Verified that this is the same as ClientOptionsExtensions.

**Review by whisper6284 — APPROVED** [review]

**Review by lucasrsant — APPROVED** [review]

**Review by danielortega-msft — APPROVED** [review]

**Pull request merged**